terraform/aws/control-plane/modules/s3/main.tf

```
@@ -2,16 +2,20 @@ data "aws_s3_bucket" "s3_conf" {
   bucket = var.name
 }
 
+locals {
+  content = merge(var.extra_content, {
+      token : var.token,
+      description : var.description,
+      locations :  [for location in var.locations : location.conf]
+      repository : length(var.private_package) > 0 ? var.private_package.conf : {}
+  })
+}
+
 resource "aws_s3_object" "conf" {
   bucket = data.aws_s3_bucket.s3_conf.id
   key    = var.object_name
   content_type = "application/json"
   content = jsonencode({
-    control-plane : {
-      token : var.token,
-      description : var.description,
-      locations :  [for location in var.locations : location.conf]
-      repository : length(var.private_package) > 0 ? var.private_package.conf : {}
-    }
+    control-plane : { for k, v in local.content : k => v if v != null }
```

not sure we should keep this null verification

```suggestion
    control-plane :  merge(var.extra_content, {
      token : var.token,
      description : var.description,
      locations :  [for location in var.locations : location.conf]
      repository : length(var.private_package) > 0 ? var.private_package.conf : {}
    })
```